spec/policies/run_policy_spec.rb

```diff
@@ -0,0 +1,75 @@
+describe RunPolicy do
+  let(:run) { create(:run) }
+  let(:admin) { create(:user, role: User::ADMIN) }
+  let(:coach) { create(:user, role: User::COACH) }
+  let(:contester) { create(:user, role: User::CONTESTER) }
+
+  [
+    :index?, :create?, :queue?, :new?
+  ].each do |action|
+    permissions action do
+
+      it "denies access to anonymous users" do
+        expect(RunPolicy).not_to permit(nil, run)
+      end
+
+      it "denies access to contesters" do
+        expect(RunPolicy).not_to permit(contester, run)
+      end
+
+      it "allows access to admins" do
+        expect(RunPolicy).to permit(admin, run)
+      end
+
+      it "allows access to coaches" do
+        expect(RunPolicy).to permit(coach, run)
+      end
+
+	  end
+  end
+
+  [
+    :update?, :edit?
+  ].each do |action|
+    permissions action do
+
+      it "denies access to anonymous users" do
+        expect(RunPolicy).not_to permit(nil, run)
+      end
+
+      it "denies access to contesters" do
+        expect(RunPolicy).not_to permit(contester, run)
+      end
+
+      it "allows access to admins" do
+        expect(RunPolicy).to permit(admin, run)
+      end
+
+      it "denies access to coaches" do
+        expect(RunPolicy).not_to permit(coach, run)
+      end
+
+    end
+  end
+
+  permissions :show? do
+    it "denies access to anonymous users" do
+      expect(RunPolicy).not_to permit(nil, run)
+    end
+
+    it "allows access to contesters if the run is visible for them" do
+      expect(RunPolicy).to permit(contester, run) if run.problem.runs_visible
```

Comment:
This is not really testing the functionality. You need to create a custom problem here which as `runs_visible = true` and validate the policy. Then the same thing for a problem with `runs_visible = false`